### PR TITLE
chore: bump peer dependencies to v17.0.2

### PR DESCRIPTION
**Pull request by alphaolomi**
Since v18 is almost out figure we can bump peer deps to v17.0.2

### CODE REVIEW

**Comment by omarnyte** [comment]
Agreed! Thanks for the contribution!

**Comment by omarnyte** [comment]
Actually, @alphaolomi is there a specific reason why we'd want to bump the React peer dependencies? [React 16.8](https://github.com/facebook/react/blob/main/CHANGELOG.md#1680-february-6-2019) introduced hooks, so I think it would be great for any apps using React 16.8 and on to be able to use this hook.

File: package.json [hunk 1]
```diff
@@ -40,8 +40,8 @@
     "typescript": "^4.3.4"
   },
   "peerDependencies": {
-    "react": "^16.8.0",
-    "react-dom": "^16.8.0"
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2"
```
Comment (omarnyte) [inline package.json]: Maybe we can use `||` to allow versions starting at 16.8.0 and  17.0.0. This is the approach used in [react-query](https://github.com/tannerlinsley/react-query/blob/master/package.json#L73) and [swr](https://github.com/vercel/swr/blob/main/package.json#L112).
```suggestion
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0"
```

**Review by omarnyte — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (alphaolomi) [reply package.json]: @omarnyte This is great idea

**Comment by alphaolomi** [comment]
@omarnyte can we ship this 🙏

**Pull request closed without merging**